backend/app/routers/files.py

```diff
@@ -345,3 +390,46 @@ async def get_file_extract(
         return {"message": "testing", "file_id": file_id}
     else:
         raise HTTPException(status_code=404, detail=f"File {file_id} not found")
+
+
+@router.post("/{file_id}/extract")
+async def resubmit_file_extractions(
+    file_id: str,
+    credentials: HTTPAuthorizationCredentials = Security(security),
+    db: MongoClient = Depends(dependencies.get_db),
+    rabbitmq_client: BlockingChannel = Depends(dependencies.get_rabbitmq),
+):
+
+    """This route will check metadata. We get the extractors run from metadata from extractors.
+    Then they are resubmitted. At present parameters are not stored. This will change once Jobs are
+    implemented.
+
+        Arguments:
+        file_idd: Id of file
+        credentials: credentials of logged in user
+        db: MongoDB Client
+        rabbitmq_client: Rabbitmq Client
+
+    """
+    if (file := await db["files"].find_one({"_id": ObjectId(file_id)})) is not None:
```

Comment:
This seems to be similar to the private method `_resubmit_file_extractors`. Can you reuse that?

Reply:
It is re-used now. Unnecessary code also taken out.